BUILD

```diff
@@ -67,9 +67,6 @@ cc_library(
     ],
     includes = ["include"],
     linkopts = select({
-        "//:osx": [
-            "-framework Foundation",
-        ],
```

Comment:
This was added in https://github.com/google/cctz/pull/102, which references https://github.com/abseil/abseil-cpp/issues/291.

Do we understand why that was wrong, or what has changed?

Reply:
From reading the history of that other PR and issue that @devbww mentioned, I think the issue was that old versions of bazel didn't automatically link with `-framework Foundation`, and newer bazel's may do that automatically. I'm guessing that the bazel versions that we support today all implicitly link w/ Foundation, so I think removing these explicit deps may work OK.

That said, I'm not sure we actually _should_ do that. Today, our code explicitly references macOS TimeZone types, for example:

https://github.com/google/cctz/blob/c674330ee6c870ff0a77e965b1811de5514bb8ae/src/time_zone_lookup.cc#L25

and in the spirit of ~include~ link with what you use, cctz probably should be explicit about its dep on `Foundation`[1], IMO.

From https://github.com/abseil/abseil-cpp/issues/326, it sounds like the real issue is w/ the native android build rules. @alyssawilk did you say that you have an alternative workaround that may allow you to fix your issue while allowing CCTZ to retain its explicit dep?

[1]: We're currently linking with `Foundation`, but we actually only need to link with `CoreFoundation` (it's a lower-level library). So this dep should probably actually be `-framework CoreFoundation`... @alyssawilk does making that change fix your issue (probably not; but we might as well check)

Reply:
Our workaround is carrying an abseil.patch file* which removes the foundation files (which looks like what everyone else is doing).  I'm now trying to get rid of that patch file, either by removing the explicit dependency or making it optional because the code is switching repos and I'm trying to clean up all the backported patches that I can.  Worst case I can probably move the backported patch but I'd rather find a fix we're all Ok with, whether it be removing the dependency as outdated and known problematic, or having an opt-in bazel select. I'm happy to do whatever works best for you folks but I'd really prefer we fix abseil rather than expect everyone downstream to patch!

https://github.com/envoyproxy/envoy-mobile/blob/main/bazel/abseil.patch

Reply:
I agree that it would be good to avoid the need for everyone (anyone) to have to apply a patch in order to use abseil/cctz w/ an android build target (when compiling on macos).

I think I'd be OK with replacing the `-framework Foundation` explicit dep with a comment stating that we actually do need the `Foundation` dep, but it's added implicitly by Bazel and not listing it explicitly works around a bug in the android native deps (and link to that issue). That's basically this PR but with an additional comment.

@devbww WDYT?

@derekmauro can you confirm whether this PR, when patched in to Abseil, will pass all of Abseil's tests?

Reply:
SG thanks!  I'f we're doing that as general clean up I'm inclined to do it for both, not just for the dependency my project needs.  I'm very happy to revert that half if y'all prefer but I'll start with that so @derekmauro 's CI run tests both removals.

Reply:
This passes CI: https://github.com/abseil/abseil-cpp/pull/1306